auth/api/iam/jar.go

```diff
@@ -163,14 +179,36 @@ func (j jar) validate(ctx context.Context, rawToken string, clientId string) (oa
 	if clientId != params.get(oauth.ClientIDParam) {
 		return nil, oauth.OAuth2Error{Code: oauth.InvalidRequestObject, Description: "invalid client_id claim in signed authorization request"}
 	}
-	// check if the signer of the JWT is the client
-	signer, err := did.ParseDIDURL(signerKid)
+	configuration, err := j.client.OpenIDConfiguration(ctx, clientId)
```

Comment:
can't remember, is this the caching HTTP client? Doesn't matter for now, might be a future TODO (nobody is going to use this in production anyways)